com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -181,24 +229,56 @@ public String[] getFullyQualifiedName(String uri, int[] lines, int[] columns) th
                 // mark it as "unverified".
                 // In future, we could consider supporting to update the breakpoint to a valid
                 // location.
-                BreakpointLocationLocator locator = new BreakpointLocationLocator(astUnit, lines[i], true,
-                        true);
+                BreakpointLocationLocator locator = new BreakpointLocationLocator(astUnit,
+                        sourceLine, true, true);
                 astUnit.accept(locator);
                 // When the final valid line location is same as the original line, that
                 // represents it's a valid breakpoint.
                 // Add location type check to avoid breakpoint on method/field which will never
                 // be hit in current implementation.
-                if (lines[i] == locator.getLineLocation()
+                if (sourceLine == locator.getLineLocation()
                         && locator.getLocationType() == BreakpointLocationLocator.LOCATION_LINE) {
-                    fqns[i] = locator.getFullyQualifiedTypeName();
+                    sourceLocation.setClassName(locator.getFullyQualifiedTypeName());
+                    if (resolvedLocations.containsKey(sourceLine)) {
+                        sourceLocation.setAvailableBreakpointLocations(resolvedLocations.get(sourceLine));
+                    } else {
+                        BreakpointLocation[] inlineLocations = getInlineBreakpointLocations(astUnit, sourceLine);
+                        sourceLocation.setAvailableBreakpointLocations(inlineLocations);
+                        resolvedLocations.put(sourceLine, inlineLocations);
+                    }
                 } else if (locator.getLocationType() == BreakpointLocationLocator.LOCATION_METHOD) {
-                    fqns[i] = locator.getFullyQualifiedTypeName().concat("#")
-                            .concat(locator.getMethodName())
-                            .concat("#").concat(locator.getMethodSignature());
+                    sourceLocation.setClassName(locator.getFullyQualifiedTypeName());
+                    sourceLocation.setMethodName(locator.getMethodName());
+                    sourceLocation.setMethodSignature(locator.getMethodSignature());
                 }
             }
         }
-        return fqns;
+
+        return sourceLocations;
+    }
+
+    private BreakpointLocation[] getInlineBreakpointLocations(final CompilationUnit astUnit, int sourceLine) {
```

Comment:
Besides the lambda expression. Do we support multiple statements in one line? for example: 

```java
int a = 0; int b = 1; int c = a+b;
```

Reply:
We only support adding an inline breakpoint for lambda only. This is also the behavior of IJ.